### PR TITLE
[css-counter-styles] Remove code for old predefined counter styles implementation

### PR DESCRIPTION
**Pull request by nt1m**
#### 6ee6584631d2851d9d76841231b74b54306fa4d3
<pre>
[css-counter-styles] Remove code for old predefined counter styles implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=256156">https://bugs.webkit.org/show_bug.cgi?id=256156</a>
rdar://108704012

Reviewed by Darin Adler.

Before this change, CSSCounterStyleAtRulesEnabled would toggle the codepath for rendering predefined (UA-level) counter styles.
This change makes rendering predefined styles go through the new codepath with CSSCounterStyleAtRulesEnabled off.
The only remaining role of setting is now to allow authors to turn on/off author-level @counter-style definitions.

Here&apos;s a summary of changes:
- Remove all the code in RenderListMarker that renders built-in counter styles when CSSCounterStyleAtRulesEnabled is off
- Always use the modern ListStyleType representation ({ ListStyleType::Type::CounterStyle, &lt;counter-style-name&gt; })
- Unconditionally parse counterStyles.css and use it to render built-in counter styles

Doing this allows easier bug fixing around pre-defined counter styles (since bugs don&apos;t need to be fixed twice), but
also removes a large amount of code.

Tested that built-in counter styles still work properly with the pref off by running:
run-webkit-tests fast/lists imported/w3c/web-platform-tests/css/css-counter-styles --experimental-feature CSSCounterStyleAtRulesEnabled=false

* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::toCSSValueID):
(WebCore::fromCSSValueID):
* Source/WebCore/editing/cocoa/FontAttributesCocoa.mm:
(WebCore::cocoaTextListMarkerName):
* Source/WebCore/rendering/RenderCounter.cpp:
(WebCore::RenderCounter::originalText const):
* Source/WebCore/rendering/RenderListMarker.cpp:
(WebCore::RenderListMarker::paint):
(WebCore::RenderListMarker::updateContent):
(WebCore::toAlphabeticOrNumeric): Deleted.
(WebCore::toSymbolic): Deleted.
(WebCore::toAlphabetic): Deleted.
(WebCore::toNumeric): Deleted.
(WebCore::toPredefinedAdditiveSystem): Deleted.
(WebCore::effectiveListMarkerType): Deleted.
(WebCore::listMarkerSuffix): Deleted.
(WebCore::listMarkerText): Deleted.
* Source/WebCore/rendering/RenderListMarker.h:
* Source/WebCore/rendering/style/ListStyleType.cpp:
(WebCore::ListStyleType::isCircle const):
(WebCore::ListStyleType::isSquare const):
(WebCore::ListStyleType::isDisc const):
(WebCore::ListStyleType::isDisclosureClosed const):
* Source/WebCore/rendering/style/ListStyleType.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::initialListStyleType):
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertListStyleType):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):

Canonical link: <a href="https://commits.webkit.org/263835@main">https://commits.webkit.org/263835@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/61a89584744356ccefb4ce5b5dd9c7e91e6b0373

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5841 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6010 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6197 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7397 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/6221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/5842 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6230 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/5974 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/8161 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/5948 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5991 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5289 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7454 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3468 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5264 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13225 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5335 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5343 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/7560 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5792 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/4754 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5230 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1385 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9352 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5591 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/rendering/RenderListMarker.cpp [hunk 1]
```diff
@@ -1697,11 +435,10 @@ void RenderListMarker::updateContent()
...
     }
-    default:
-        auto text = listMarkerText(styleType.type, m_listItem->value());
-        m_textWithSuffix = makeString(text, listMarkerSuffix(styleType.type));
-        m_textWithoutSuffixLength = text.length();
-        m_textIsLeftToRightDirection = u_charDirection(text[0]) != U_RIGHT_TO_LEFT;
+    case ListStyleType::Type::None:
+        m_textWithSuffix = " "_s;
+        m_textWithoutSuffixLength = 0;
+        m_textIsLeftToRightDirection = false;
```
Comment (nt1m) [inline Source/WebCore/rendering/RenderListMarker.cpp]: This should be `true`, will fix this in the next revision

File: Source/WebCore/rendering/RenderListMarker.cpp [hunk 2]
```diff
@@ -43,1272 +43,10 @@
 
 namespace WebCore {
 
-// As of June 2021, the Microsoft C++ compiler seems unable to include std::initializer_list in a constexpr.
```
Comment (vitorroriz) [inline Source/WebCore/rendering/RenderListMarker.cpp:46]: is this change related?

File: Source/WebCore/rendering/RenderListMarker.cpp [hunk 3]
```diff
@@ -43,1272 +43,10 @@
...
-#if COMPILER(MSVC)
-#define CONSTEXPR_WITH_MSVC_INITIALIZER_LIST_WORKAROUND const
-#else
-#define CONSTEXPR_WITH_MSVC_INITIALIZER_LIST_WORKAROUND constexpr
-#endif
-
```
Comment (vitorroriz) [inline Source/WebCore/rendering/RenderListMarker.cpp:54]: is this change related?

Comment (nt1m) [reply Source/WebCore/rendering/RenderListMarker.cpp:54]: Yes, we're no longer using CONSTEXPR_WITH_MSVC_INITIALIZER_LIST_WORKAROUND throughout the file, so we can remove the definition

Comment (vitorroriz) [reply Source/WebCore/rendering/RenderListMarker.cpp:54]: Nevermind, it seems we won't need that anymore.

**Review by darinadler — APPROVED** [review]

File: Source/WebCore/style/StyleBuilderConverter.h [hunk 4]
```diff
@@ -298,14 +298,15 @@ inline ListStyleType BuilderConverter::convertListStyleType(const BuilderState&
     if (primitiveValue.isValueID()) {
         if (primitiveValue.valueID() == CSSValueNone)
             return { ListStyleType::Type::None, nullAtom() };
-        if (builderState.document().settings().cssCounterStyleAtRulesEnabled())
-            return { ListStyleType::Type::CounterStyle, makeAtomString(primitiveValue.stringValue()) };
-        return { fromCSSValue<ListStyleType::Type>(primitiveValue), nullAtom() };
+        return { ListStyleType::Type::CounterStyle, makeAtomString(primitiveValue.stringValue()) };
```
Comment (vitorroriz) [inline Source/WebCore/style/StyleBuilderConverter.h:301]: What happens during parsing if user turns this flags off?

Comment (vitorroriz) [reply Source/WebCore/style/StyleBuilderConverter.h:301]: I believe when parsing the descriptors with context in `UASheetMode` we set `propertySettings.cssCounterStyleAtRulesEnabled` to true so we should be fine. But can you double check, please?

Comment (nt1m) [reply Source/WebCore/style/StyleBuilderConverter.h:301]: Yes, and also `consumeListStyleType` consumes predefined value IDs unconditionally, so nothing changes here.
